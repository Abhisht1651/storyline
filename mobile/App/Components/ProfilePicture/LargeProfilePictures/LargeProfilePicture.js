// @flow

import React from 'react'
import { TouchableOpacity, Image, View } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import Images from '../../../Themes/Images'
import styles from '../../Styles/ProfilePictureStyles/LargeProfilePictureStyles/LargeProfilePictureStyle'

type Props = {
  image: Image,
  type: number,
  onPress: () => {}
}

const LargeProfilePicture = (props: Props) => {
  const { image, type, onPress } = props
  let component = {}

  if (type === 2) {
    component = (
      <LinearGradient
        colors={['#46C3F4', '#B27CF3']}
        style={styles.linearGradient}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}>
        <TouchableOpacity style={styles.container2} activeOpacity={1} onPress={onPress}>
          <Image source={image} style={styles.image2} />
        </TouchableOpacity>
      </LinearGradient>
    )
  } else if (type === 3) {
    component = (
      <TouchableOpacity activeOpacity={1} style={styles.imageStyle3} onPress={onPress}>
        <Image source={image} style={styles.imageStyle3} />
        <View activeOpacity={1} style={styles.online} />
      </TouchableOpacity>
    )
  } else if (type === 4) {
    component = (
      <LinearGradient
        colors={['#46C3F4', '#B27CF3']}
        style={styles.linearGradient}
        start={{ x: 0, y: 0 }}
        end={{ x: 0, y: 1 }}>
        <TouchableOpacity activeOpacity={1} style={styles.container4} onPress={onPress}>
          <Image source={image} style={styles.imageStyle4} />
        </TouchableOpacity>
        <View activeOpacity={1} style={styles.online} />
      </LinearGradient>
    )
  } else {
    component = (
      <TouchableOpacity style={styles.container1} activeOpacity={1} onPress={onPress}>
        <Image source={image} style={styles.image1} />
      </TouchableOpacity>
    )
  }

  return <View>{component}</View>
}
LargeProfilePicture.defaultProps = {
  image: Images.profilePictureSmall,
  type: 4,
  onPress: () => {}
}

export default LargeProfilePicture
