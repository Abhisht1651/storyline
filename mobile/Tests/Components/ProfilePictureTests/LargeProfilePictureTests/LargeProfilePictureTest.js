import { shallow } from 'enzyme'
import React from 'react'
import LargeProfilePicture from '../../../../App/Components/ProfilePicture/LargeProfilePictures/LargeProfilePicture'

describe('LargeProfilePicture', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<LargeProfilePicture />)
      expect(component).toMatchSnapshot()
    })
  })
})
