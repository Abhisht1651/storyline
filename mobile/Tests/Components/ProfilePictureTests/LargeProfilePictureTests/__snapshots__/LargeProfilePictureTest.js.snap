// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeProfilePicture Rendering should match to snapshot 1`] = `
<View>
  <LinearGradient
    colors={
      Array [
        "#46C3F4",
        "#B27CF3",
      ]
    }
    end={
      Object {
        "x": 0,
        "y": 1,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "aspectRatio": 1,
        "borderRadius": 100,
        "height": 105.14285714285714,
      }
    }
  >
    <ForwardRef
      activeOpacity={1}
      onPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "flex": 1,
          "justifyContent": "center",
          "margin": 2,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 2,
            "height": "100%",
          }
        }
      />
    </ForwardRef>
    <View
      activeOpacity={1}
      style={
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#47C3F4",
          "borderColor": "#FFFFFF",
          "borderRadius": 100,
          "borderWidth": 2,
          "bottom": 0,
          "height": 10,
          "justifyContent": "center",
          "left": 70.64285714285714,
          "position": "absolute",
        }
      }
    />
  </LinearGradient>
</View>
`;
